vm/src/stdlib/os.rs

```diff
@@ -157,15 +159,25 @@ impl TryFromObject for PyPathLike {
     }
 }
 
-fn make_path<'a>(
-    vm: &VirtualMachine,
-    path: &'a PyPathLike,
-    dir_fd: &DirFd,
-) -> PyResult<&'a ffi::OsStr> {
-    if dir_fd.0.is_some() {
-        Err(vm.new_os_error("dir_fd not supported yet".to_owned()))
-    } else {
-        Ok(path.path.as_os_str())
+fn make_path(vm: &VirtualMachine, path: &PyPathLike, dir_fd: &DirFd) -> PyResult<ffi::OsString> {
```

Comment:
You should be able to return `std::borrow::Cow<ffi::OsStr>`, since unless dir_fd is `Some` you can just borrow from `path`

Reply:
Thanks for reviewing the code. It is fixed in commits below.